src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -3154,7 +3167,7 @@ internal BoundExpression CreateReturnConversion(
 
             diagnostics.Add(syntax, useSiteInfo);
 
-            if (!argument.HasAnyErrors)
+            if (!argument.HasAnyErrors || argument.Kind == BoundKind.UnboundLambda)
```

Comment:
>!argument.HasAnyErrors || argument.Kind == BoundKind.UnboundLambda [](http://example.com/codeflow?start=16&length=66)

It looks like according to PR description this change is motivated by consistency with `BindYieldReturnStatement`. However, the conditions do not look the same to me. My preference would be to not touch this code, or use the same condition. Specifically ```!(argument.HasErrors || ((object)argument.Type != null && argument.Type.IsErrorType()))```